### PR TITLE
Basic code for a new trends UI layout

### PR DESCRIPTION
**Pull request by jdhollen**
More to come (this is all flag-guarded), but I want to get the basic skeleton in before I do more work on the overview page, etc.

It looks... like this:
Screenshot 2023-08-07 at 1 43 25 PM
And this:
Screenshot 2023-08-07 at 1 57 25 PM

**Related issues**: N/A

### CODE REVIEW

**Comment by siggisim** [comment]
Will the overview page just have those 3 boxes? Or is there more stuff we want to show there?

Currently seems a bit bare - I always wonder if people will actually ever click on the tabs

**Comment by jdhollen** [comment]
> Will the overview page just have those 3 boxes? Or is there more stuff we want to show there?
> 
> Currently seems a bit bare - I always wonder if people will actually ever click on the tabs

Yes--lots more to do here, just wanted to get the overall structure in so that i'm not sending out a 5000-line change or whatever.  This'll let me do single charts / tables for subsequent changes.

File: enterprise/app/trends/trends_request_helper.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+type TrendsDataUpdateCallback = (model: TrendsModel) => void;
+
+export enum TrendsRequestType {
+  GET_TRENDS,
+  GET_HEATMAP,
+}
+
+export default class TrendsRequestHelper {
```
[the remark this answers is not in the document]
Comment (jdhollen) [reply enterprise/app/trends/trends_request_helper.ts]: Sure, done, sorry for leaving this sitting.

**Comment by jdhollen** [comment]
rebased for my own sake.

**Review by bduffany — APPROVED** [review]

**Pull request merged**